### PR TITLE
fix(components): align badge style with guidelines

### PR DESCRIPTION
**Pull request by talend-sromain**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

Badges are not aligned with our guidelines

**What is the new behavior?**

Badges render is closer to guidelines

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://369.talend.surge.sh)

File: packages/components/src/Badge/Badge.scss [hunk 1]
```diff
@@ -10,23 +10,27 @@
...
 		svg {
-			color: $black;
+			color: $text-color;
 		}
 
 		&.tc-badge-with-icon:hover {
 			background: opacify($brand-primary, 0.25);
+			color: $white;
+
+			svg {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss]: Please, refactor here: 
```scss
&,
svg {
  color: $white;
}
```

File: packages/components/src/Badge/Badge.scss [hunk 2]
```diff
@@ -10,23 +10,27 @@
 	&.tc-badge-outline {
 		background: $white;
 		border: 1px solid $brand-primary-darker;
-		color: $black;
+		color: $text-color;
 		font-weight: normal;
 
 		svg {
```
Comment (frassinier) [inline packages/components/src/Badge/Badge.scss]: Please, refactor here:
```scss
&,
svg {
  color: $text-color;
}
```

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://369.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://369.talend.surge.sh)

**Pull request merged**